common/djangoapps/third_party_auth/api/utils.py

```diff
@@ -0,0 +1,29 @@
+"""
+Shareable utilities for third party auth api functions
+"""
+
+
+def filter_user_social_auth_queryset_by_provider(query_set, provider):
+    """
+    Filter a query set by the given TPA provider
+
+    Params:
+        query_set: QuerySet[UserSocialAuth]
+        provider: common.djangoapps.third_party_auth.models.ProviderConfig
+    Returns:
+        QuerySet[UserSocialAuth]
+    """
+    # Note: When using multi-IdP backend, the provider column isn't
+    # enough to identify a specific backend
+    filtered_query_set = query_set.filter(provider=provider.backend_name)
+
+    # Test if the current provider has a slug which it appends to
+    # uids; these can be used to identify the backend more
+    # specifically than the provider's backend
+    fake_uid = 'uid'
+    uid = provider.get_social_auth_uid(fake_uid)
+    if uid != fake_uid:
+        # if yes, we add a filter for the slug on uid column
+        filtered_query_set = filtered_query_set.filter(uid__startswith=uid[:-len(fake_uid)])
```

Comment:
Might not matter, but what if `uid` is shorter than 3 characters?

Reply:
`uid` will either be `fake_uid` or `fake_uid` prepended with the slug of the provider we're trying to identify. I rewrote this specifically to make more clear that this is carving off the known, constant-length bit from the end. Does that help?

Reply:
👍 Yes, very helpful.  If you're making changes anyway, I'd find it helpful to have what you just said in a comment above.  If you're not making any other changes, don't worry about it.